agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2123,6 +2160,69 @@ backhaul_manager::get_sta_radio(const sMacAddr &sta_mac)
     return radio != slaves_sockets.end() ? *radio : nullptr;
 }
 
+bool backhaul_manager::handle_multi_ap_policy_config_request(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                             const std::string &src_mac)
+{
+    auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received MULTI_AP_POLICY_CONFIG_REQUEST_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto steering_policy_tlv = cmdu_rx.getClass<wfa_map::tlvSteeringPolicy>();
+    if (steering_policy_tlv) {
+        // For the time being, agent doesn't do steering so steering policy is ignored.
+    }
+
+    auto metric_reporting_policy_tlv = cmdu_rx.getClass<wfa_map::tlvMetricReportingPolicy>();
+    if (metric_reporting_policy_tlv) {
+        /**
+         * The Multi-AP Policy Config Request message containing a Metric Reporting Policy TLV is
+         * sent by the controller and received by the backhaul manager.
+         * The backhaul manager forwards the request message "as is" to all the slaves managing the
+         * radios which Radio Unique Identifier has been specified.
+         */
+        for (size_t i = 0; i < metric_reporting_policy_tlv->metrics_reporting_conf_list_length();
+             i++) {
+            auto tuple = metric_reporting_policy_tlv->metrics_reporting_conf_list(i);
+            if (!std::get<0>(tuple)) {
+                LOG(ERROR) << "Failed to get metrics_reporting_conf[" << i
+                           << "] from TLV_METRIC_REPORTING_POLICY";
+                return false;
+            }
+
+            auto metrics_reporting_conf = std::get<1>(tuple);
+
+            std::shared_ptr<sRadioInfo> radio = get_radio(metrics_reporting_conf.radio_uid);
+            if (radio) {
+                uint16_t length = message_com::get_uds_header(cmdu_rx)->length;
+                cmdu_rx.swap(); // swap back before forwarding
+                if (!message_com::forward_cmdu_to_uds(radio->slave, cmdu_rx, length)) {
```

Comment:
I see this idiom for the second time in this PR, maybe it should be moved to a separate function.

Reply:
This duplicate code will disappear with Unified Agent as there will be no need for such message forwarding.